src/Microsoft.TemplateEngine.Core/Expressions/Cpp/CppStyleEvaluatorDefinition.cs

```diff
@@ -113,6 +116,8 @@ public static bool CppStyleEvaluator(IProcessorState processor, ref int bufferLe
             }
 
             bool first = true;
+            Stack<TokenFamily> quoteNesting = new Stack<TokenFamily>();
```

Comment:
Quotes in quotes shouldn't have to be tracked (presumably for balancing?) - ex `"they're"` is perfectly OK. We should just maintain a "is in quoting mode and here's my original open quote type" to determine when the block ends (a quote matching the original)